### PR TITLE
Ee 000 update tap GitHub schema

### PR DESCRIPTION
**Pull request by rafael-lima-tw**
## Problem

We need to guarantee that the tap github contract is being broken

## Proposed changes

Add json schema validation for tap-github

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] Description above provides context of the change
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Unit tests for changes (not needed for documentation changes)
- [x] CI checks pass with my changes
- [x] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [x] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [x] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions

### CODE REVIEW

**Review by diegofurtados — APPROVED** [review]

File: pipelinewise/cli/schemas/tap.json [hunk 1]
```diff
@@ -232,6 +232,46 @@
...
+            },
+            "include_archived": {
+              "type": "boolean",
+              "default": false
+            },
+            "include_disabled": {
+              "type": "boolean",
+              "default": false
+            },
+            "max_rate_limit_wait_seconds": {
```
Comment (Samira-El) [inline pipelinewise/cli/schemas/tap.json:268]: Can you set a sane min and max for this parameter?

File: pipelinewise/cli/schemas/tap.json [hunk 2]
```diff
@@ -232,6 +232,46 @@
...
+    },
+    "tap_github": {
+      "properties": {
+        "type": {
+          "enum": ["tap_github"]
+        },
+        "db_conn": {
+          "type": "object",
+          "required": ["access_token"],
+          "properties": {
```
Comment (Samira-El) [inline pipelinewise/cli/schemas/tap.json:244]: `start_date` is missing, it's also a required property

File: pipelinewise/cli/schemas/tap.json [hunk 3]
```diff
@@ -232,6 +232,46 @@
         }
       },
       "required": ["type"]
+    },
+    "tap_github": {
+      "properties": {
+        "type": {
+          "enum": ["tap_github"]
+        },
+        "db_conn": {
```
Comment (Samira-El) [inline pipelinewise/cli/schemas/tap.json:241]: Can you add conditions so that either `repository` or `organization` is present?

**Review by Samira-El — CHANGES_REQUESTED** [review]
Left few comments.

Can you also please add unit tests for this validation, they should be located [here](https://github.com/transferwise/pipelinewise/blob/master/tests/units/cli/test_cli_utils.py)

**Review by beveradb — DISMISSED** [review]

**Review by beveradb — DISMISSED** [review]

**Review by diegofurtados — APPROVED** [review]

File: pipelinewise/cli/schemas/tap.json [hunk 4]
```diff
@@ -232,6 +232,84 @@
...
+                },
+                {
+                  "if": {
+                    "const": "repository"
+                  },
+                  "then": {
+                    "not": { "required": ["organization"] }
+                  }
+                }
+              ]
```
Comment (Samira-El) [inline pipelinewise/cli/schemas/tap.json:308]: I don't this block is doing anything tbh, I ran it against some scenarios and it passed when it should have failed.
Better to remove it than keep it.

**Review by Samira-El — APPROVED** [review]

**Pull request merged**